server/cluster/region_stat.go

```diff
@@ -0,0 +1,94 @@
+// Copyright 2019 TiKV Project Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cluster
+
+import (
+	"time"
+
+	"github.com/tikv/pd/server/core"
+	"github.com/tikv/pd/server/schedule"
+)
+
+// clearThreshold indicates when we do cleanup operations.
+// When the number of times the collect() function is executed is exceeds this threshold,
+// regions that are not in region tree will be deleted.
+const clearThreshold = 10
+
+// regionStateType represents the type of region's state.
+type regionStateType int
+
+// region state type
+const (
+	regionStateDown regionStateType = iota
```

Comment:
Consider replacing `Down` with `NoLeader` to prevent confusion with `down-peer`?